### ISSUE TITLE
Improve multi-class model reloading

### ISSUE THREAD

**Issue by bw4sz** [weecology/DeepForest#287]
Loading a multi-class model requires you to know the label dict and preset the classes, this is cumbersome.

If you use main.save_model and are loading a multi_class model, it requires you to create a new object, since main.deepforest() instantiates a deepforest single class model by default.

```
TRAINED_MODEL = "/blue/ewhite/everglades/Zooniverse/20211215_112228/species_model.pl"
train = pd.read_csv("/blue/ewhite/everglades/Zooniverse/parsed_images/species_train.csv")
label_dict = {key:value for value, key in enumerate(train.label.unique())}
m = main.deepforest(num_classes=len(label_dict), label_dict=label_dict)
m.load_state_dict(torch.load(TRAINED_MODEL, map_location="cpu")["state_dict"])
```

In the mean time the docs need to be clearer.

**Comment by ethanwhite** [comment]
:+1: I've been running into this issue as well.

**Comment by bw4sz** [comment]
Added to docs.

Loading a multi_class model which was saved using main.save_model is currently somewhat cumbersome (see https://github.com/weecology/DeepForest/issues/287). 
The easiest way is to create a new deepforest object and preset the number of classes and label dict, just as you did during training.

```
import pandas as pd

TRAINED_MODEL = "/blue/ewhite/everglades/Zooniverse/20211215_112228/species_model.pl"
train = pd.read_csv("/blue/ewhite/everglades/Zooniverse/parsed_images/species_train.csv")
label_dict = {key:value for value, key in enumerate(train.label.unique())}
m = main.deepforest(num_classes=len(label_dict), label_dict=label_dict)
m.load_state_dict(torch.load(TRAINED_MODEL, map_location="cpu")["state_dict"])
```

**Comment by ethanwhite** [comment]
Yeah, I had done what you did, just echoing that I think making it easier is important.

**Comment by bw4sz** [comment]
https://pytorch-lightning.readthedocs.io/en/latest/common/hyperparameters.html is somewhere in the right direction.
